data/scripts/talkactions/player/reward.lua

```diff
@@ -0,0 +1,62 @@
+local config = {
+	items = {
+		{ id = 35284, charges = 64000 },
+		{ id = 35279, charges = 64000 },
+		{ id = 35281, charges = 64000 },
+		{ id = 35283, charges = 64000 },
+		{ id = 35282, charges = 64000 },
+		{ id = 35280, charges = 64000 },
+	},
+	storage = tonumber(Storage.PlayerWeaponReward), -- storage key, player can only win once
+}
+
+local function sendExerciseRewardModal(player)
+	local window = ModalWindow {
+		title = "Exercise Reward",
+		message = 'choose a item'
+	}
+	for _, it in pairs(config.items) do
+		local iType = ItemType(it.id)
+		if iType then
+			window:addChoice(iType:getName(), function(player, button, choice)
+				if button.name ~= "Select" then
+					return true
+				end
+
+				local inbox = player:getSlotItem(CONST_SLOT_STORE_INBOX)
+				if inbox and inbox:getEmptySlots() > 0 then
+					local item = inbox:addItem(it.id, it.charges)
+					if item then
+						item:setActionId(IMMOVABLE_ACTION_ID)
```

Comment:
item:setAttribute(ITEM_ATTRIBUTE_STORE, systemTime())

Reply:
use after `item:setActionId(IMMOVABLE_ACTION_ID)` or replace?

Reply:
Replace. That way players can store their rewards in depot.